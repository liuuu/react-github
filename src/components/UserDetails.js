import React from "react";
import { Card, Avatar, Col, Row, Spin } from "antd";
import { Link } from "react-router-dom";

const UserDetails = ({ detail }) =>
  <div className="user-detail">
    <Row type="flex" justify="center" style={{ marginTop: 50 }}>
      <Col span={10}>
        <Card bodyStyle={{ padding: 0 }} bordered={false}>

          <ul className="space-list-item">
            <li>
              <img
                style={{ width: "70%" }}
                className="avatar"
                src={detail.avatar_url}
                alt={detail.login}
              />
            </li>
            <li><p>{detail.name}</p></li>

            <li>{detail.bio}</li>
            <li><Link to="/all">back to all</Link></li>

          </ul>
        </Card>
      </Col>
    </Row>
  </div>;

export default UserDetails;
