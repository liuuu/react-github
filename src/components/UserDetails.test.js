import React from "react";
import UserDetails from "./UserDetails";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import { MemoryRouter } from "react-router-dom";
const props = {
  detail: {
    avatar_url: "any",
    login: "any",
    name: "any",
    bio: "any"
  }
};
it("should match snapshot", () => {
  const tree = render(
    <MemoryRouter>
      <UserDetails {...props} />
    </MemoryRouter>
  );
  expect(toJson(tree)).toMatchSnapshot();
});
