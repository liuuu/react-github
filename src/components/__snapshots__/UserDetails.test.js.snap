// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  class="user-detail"
>
  <div
    class="ant-row-flex ant-row-flex-center"
    style="margin-top:50px;"
  >
    <div
      class="ant-col-10"
    >
      <div
        class="ant-card"
      >
        <div
          class="ant-card-body"
          style="padding:0;"
        >
          <ul
            class="space-list-item"
          >
            <li>
              <img
                alt="any"
                class="avatar"
                src="any"
                style="width:70%;"
              />
            </li>
            <li>
              <p>
                any
              </p>
            </li>
            <li>
              any
            </li>
            <li>
              <a
                href="/all"
              >
                back to all
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
